### DISCUSSION TITLE
Detailed guide or documentations for SELECT ?

### DISCUSSION THREAD

**Discussion by keepsimple1** [diesel-rs/diesel#2735] [Q&A]
Hi, in the [Diesel guides](https://diesel.rs/guides/), there are separate chapters for `UPDATE` and `INSERT`, but I cannot find detailed chapters for `SELECT`. Any pointers?

Particularly, I'm looking for ways of doing `SELECT` with multiple filters (i.e. filter on multiple columns).

**Comment by weiznich** [comment]
There is currently no complete guide for select statements. We are waiting for someone with some time to write such a guide.
That written: All methods that can be used to construct a select statement are documented [here]( https://docs.diesel.rs/1.4.x/diesel/query_dsl/trait.QueryDsl.html). You likely want to use one of the filter methods.

**Reply by keepsimple1** [reply]
Thanks. I have a follow-up question:  how do we use `filter` methods to implement SQL `IN` operator? I could not find docs or examples for that.  

SQL example of `IN`: 
```sql
WHERE column_name IN (value1, value2, ...);
```

**Reply by keepsimple1** [reply]
I think I found it. `eq_any()` seems to be.
